src/protocol/ipa_prf/boolean_ops/comparison_and_subtraction_low_com.rs

```diff
@@ -0,0 +1,439 @@
+                        ctx.set_total_records(1),
+                        protocol::RecordId(0),
+                        &x_y[0],
+                        &x_y[1],
+                    )
+                    .await
+                    .unwrap()
+                })
+                .await
+                .reconstruct();
+
+            assert_eq!(result, <Boolean>::from(expected));
+
+            let result2 = world
+                .semi_honest(records.into_iter(), |ctx, x_y| async move {
+                    compare_geq::<_, BA64, BA64>(
+                        ctx.set_total_records(1),
+                        protocol::RecordId(0),
+                        &x_y[0],
+                        &x_y[0],
+                    )
+                    .await
+                    .unwrap()
+                })
+                .await
+                .reconstruct();
+            assert_eq!(result2, <Boolean>::from(true));
+        });
+    }
+
+    ///testing comparisons gt
+    #[test]
+    fn semi_honest_compare_gt() {
+        run(|| async move {
+            let world = TestWorld::default();
+
+            let mut rng = thread_rng();
+
+            let records: Vec<BA64> = vec![rng.gen::<BA64>(), rng.gen::<BA64>()];
+            let x = records[0].as_u128();
+            let y = records[1].as_u128();
+
+            let expected = x >= y;
+
+            let result = world
+                .semi_honest(records.clone().into_iter(), |ctx, x_y| async move {
+                    compare_gt::<_, BA64, BA64>(
+                        ctx.set_total_records(1),
+                        protocol::RecordId(0),
+                        &x_y[0],
+                        &x_y[1],
+                    )
+                    .await
+                    .unwrap()
+                })
+                .await
+                .reconstruct();
+
+            assert_eq!(result, <Boolean>::from(expected));
+
+            let result2 = world
+                .semi_honest(records.into_iter(), |ctx, x_y| async move {
+                    compare_gt::<_, BA64, BA64>(
+                        ctx.set_total_records(1),
+                        protocol::RecordId(0),
+                        &x_y[0],
+                        &x_y[0],
+                    )
+                    .await
+                    .unwrap()
+                })
+                .await
+                .reconstruct();
+            assert_eq!(result2, <Boolean>::from(false));
+        });
+    }
+
+    ///testing correctness of subtraction
+    #[test]
+    fn semi_honest_sub() {
+        run(|| async move {
+            let world = TestWorld::default();
+
+            let mut rng = thread_rng();
+
+            let records: Vec<BA64> = vec![rng.gen::<BA64>(), rng.gen::<BA64>()];
+            let x = records[0].as_u128();
+            let y = records[1].as_u128();
+
+            let expected = ((x + 1 + u128::from(u64::MAX)) - y) % (1 + u128::from(u64::MAX));
```

Comment:
I think this is another good opportunity to use the 
```
let z = 1_u128 << 64;
```

Then this line becomes:
```
let expected = ((x + z) - y) % z;
```